parser.go

```diff
@@ -975,7 +975,21 @@ func processRouterOperation(parser *Parser, operation *Operation) error {
 			parser.debug.Printf("warning: %s\n", err)
 		}
 
-		*op = &operation.Operation
+		if len(operation.RouterProperties) > 1 {
+			newOp := *operation
+			validParams := []spec.Parameter{}
```

Comment:
Please use `var validParams []spec.Parameter` in order to avoid mem allocations.

Reply:
Updated